corehq/apps/settings/forms.py

```diff
@@ -54,8 +56,15 @@ def __init__(self, user, *args, **kwargs):
             ),
         )
 
-    def clean_new_password1(self):
-        return clean_password(self.cleaned_data.get('new_password1'))
+    def clean_old_password(self):
+        from corehq.apps.hqwebapp.utils import decode_password
+        old_password = decode_password(self.cleaned_data['old_password'])
+        if old_password == '':
+            raise ValidationError(
+                _("Password cannot be empty"), code='password_empty',
+            )
```

Comment:
Yes that is correct.

This check would only run in case the encoding is incorrect/messed up with, since one can not enter blank password because the front end does not allow. Hence, the user would get a message password empty and not password incorrect which would be a better message for the user that we are getting empty password and not incorrect password.